.github/workflows/build_develop.yml

```
@@ -34,28 +38,81 @@ jobs:
           SENTRY_URL: ${{ secrets.SENTRY_URL }}
           SENTRY_ORG: element
           SENTRY_PROJECT: riot-web
+          # We have to update the CSP until Cloudflare Pages supports proxying redirects
+          CSP_EXTRA_SOURCE: ${{ env.R2_PUBLIC_URL }}
 
       - run: mv dist/element-*.tar.gz dist/develop.tar.gz
 
-      # We keep the latest develop.tar.gz as the artifact uploaded later expires after 24 and requires auth to download
-      # Element Desktop's fetch script uses this tarball to fetch latest develop to build Nightlies.
-      - name: Deploy develop.tar.gz to Github Pages
-        uses: JamesIves/github-pages-deploy-action@v4
-        with:
-          folder: dist
-          single-commit: true
+      # TODO: remove this once the element-desktop fetch script is updated to grab develop.element.io/develop.tar.gz
+#      - name: Deploy develop.tar.gz to Github Pages
+#        uses: JamesIves/github-pages-deploy-action@v4
+#        with:
+#          folder: dist
+#          single-commit: true
 
       - uses: actions/upload-artifact@v3
         with:
           name: webapp
           path: dist/develop.tar.gz
           retention-days: 1
+          
+      - name: Extract webapp
+        run: |
+          mkdir _deploy
+          tar xf dist/develop.tar.gz -C _deploy --strip-components=1
+          
+      - name: Copy config
+        run: cp element.io/develop/config.json _deploy/config.json
+
+      - name: Populate 404.html
+        run: echo "404 Not Found" > _deploy/404.html
+
+      - name: Populate _headers
+        run: cp .github/cfp_headers _deploy/_headers
+
+      - name: Populate _redirects
+        run: |
+          echo "/develop.tar.gz $R2_PUBLIC_URL/develop.tar.gz 301" >> _deploy/_redirects
+          for bundle in $(aws s3 ls s3://$R2_BUCKET/bundles/ --endpoint-url $R2_URL --region=auto | awk '{print $2}'); do
+            echo "/bundles/${bundle}* $R2_PUBLIC_URL/bundles/${bundle}:splat 301" >> _deploy/_redirects
+          done
+          cat _deploy/_redirects
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.CF_R2_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.CF_R2_TOKEN }}
 
       - name: Wait for other steps to succeed
         uses: lewagon/wait-on-check-action@v1.1.2
         with:
           ref: ${{ github.ref }}
-          running-workflow-name: 'Build & Upload source maps to Sentry'
+          running-workflow-name: 'Build & Deploy develop.element.io'
           repo-token: ${{ secrets.GITHUB_TOKEN }}
           wait-interval: 10
           check-regexp: ^((?!SonarQube|issues|board).)*$
+
+      # We keep the latest develop.tar.gz as the artifact uploaded expires after 24h and requires auth to download
```

this is incomprehensible. Is the following accurate?

```suggestion
      # We keep the latest develop.tar.gz (as well as all the bundles) on R2, as 
      # GHA artifacts expire after 24h and require auth to download.
```